src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -260,14 +261,28 @@ CreateReferenceTable(Oid relationId)
  */
 static void
 ConvertToDistributedTable(Oid relationId, char *distributionColumnName,
-						  char distributionMethod, uint32 colocationId,
-						  char replicationModel)
+						  char distributionMethod, uint32 colocationId)
 {
 	Relation relation = NULL;
 	TupleDesc relationDesc = NULL;
 	char *relationName = NULL;
 	char relationKind = 0;
 	Var *distributionColumn = NULL;
+	char replicationModel = REPLICATION_MODEL_INVALID;
+
+	if (ReplicationModel == REPLICATION_MODEL_STREAMING && ShardReplicationFactor != 1)
+	{
+		ereport(ERROR, (errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+						errmsg("invalid replication settings"),
+						errdetail("Replication factors greater than one are incompatible "
+								  "with the streaming replication model."),
+						errhint("Try again after reducing \"citus.shard_replication_"
+								"factor\" to one or setting \"citus.replication_model\" "
+								"to \"statement\".")));
```

Comment:
Feel free to ignore: I think this error message of moving all actually useful things in an error message into detail. I.e. this message would be a lot more useful if it, slightly rephrased, contained the detail instead of the primary message.